packages/hub/src/graphql/types/Model.ts

```diff
@@ -2,8 +2,41 @@ import { builder } from "@/graphql/builder";
 import { ModelRevision, ModelRevisionConnection } from "./ModelRevision";
 import { decodeGlobalID } from "@pothos/plugin-relay";
 
+import { Session } from "next-auth";
+import { Prisma } from "@prisma/client";
+
+export function modelWhereHasAccess(
+  session: Session | null
+): Prisma.ModelWhereInput {
+  return {
+    OR: [
+      { isPrivate: false },
+      ...(session
+        ? [
+            {
+              owner: {
+                email: session.user.email,
+              },
+            },
+          ]
+        : []),
+    ],
+  };
+}
+
 export const Model = builder.prismaNode("Model", {
   id: { field: "id" },
+  include: {
+    owner: true, // required for auth
+  },
+  authScopes: (model, context) => {
+    if (!model.isPrivate || context.session?.user.email === model.owner.email) {
```

Comment:
i'd probably separate some of this into a separate function, like ``isOwner = () => context.session?.user.email === model.owner.email``. Very minor though.